Fix test_public_bindings to skip examples

aorenste opened the pull request:
Test:
```
python test/test_public_bindings.py -k TestPublicBindings.test_correct_module_names
```
before this change it fails with an error in `parse_args` ending with a SystemExit and a stack trace because torch/distributed/_tensor/examples/comm_mode_features_example_argparser.py runs parsing code at top-level (assuming it will be '__main__') - so when the test imports that file it fails.

After this change it succeeds.

Also added `__all__` to a few files so it would succeed cleanly.



Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127596
* #129780
* #128041
* #129182
* __->__ #129524

aorenste committed:
Fix test_public_bindings to skip examples

[ghstack-poisoned]

aorenste committed:
Update on "Fix test_public_bindings to skip examples"

Test:
```
python test/test_public_bindings.py -k TestPublicBindings.test_correct_module_names
```
before this change it fails with an error in `parse_args` ending with a SystemExit and a stack trace because torch/distributed/_tensor/examples/comm_mode_features_example_argparser.py runs parsing code at top-level (assuming it will be '__main__') - so when the test imports that file it fails.

After this change it succeeds.

Also added `__all__` to a few files so it would succeed cleanly.






[ghstack-poisoned]

albanD commented:
Why do we have example code in the package we ship to our end users? I would expect it to be in our examples repo: https://github.com/pytorch/examples ?

aorenste commented:
> Why do we have example code in the package we ship to our end users? I would expect it to be in our examples repo: [pytorch/examples](https://github.com/pytorch/examples) ?

It's likely that these aren't shipped to the end users (I'm not sure what gets ignored when packaging or not) - but it's useful to be able to run test_public_bindings locally after it fails on CI and locally it complains about these files.

aorenste committed:
Update on "Fix test_public_bindings to skip examples"

Test:
```
python test/test_public_bindings.py -k TestPublicBindings.test_correct_module_names
```
before this change it fails with an error in `parse_args` ending with a SystemExit and a stack trace because torch/distributed/_tensor/examples/comm_mode_features_example_argparser.py runs parsing code at top-level (assuming it will be '__main__') - so when the test imports that file it fails.

After this change it succeeds.

Also added `__all__` to a few files so it would succeed cleanly.






[ghstack-poisoned]

aorenste committed:
Update on "Fix test_public_bindings to skip examples"

Test:
```
python test/test_public_bindings.py -k TestPublicBindings.test_correct_module_names
```
before this change it fails with an error in `parse_args` ending with a SystemExit and a stack trace because torch/distributed/_tensor/examples/comm_mode_features_example_argparser.py runs parsing code at top-level (assuming it will be '__main__') - so when the test imports that file it fails.

After this change it succeeds.

Also added `__all__` to a few files so it would succeed cleanly.






[ghstack-poisoned]

aorenste committed:
Update on "Fix test_public_bindings to skip examples"

Test:
```
python test/test_public_bindings.py -k TestPublicBindings.test_correct_module_names
```
before this change it fails with an error in `parse_args` ending with a SystemExit and a stack trace because torch/distributed/_tensor/examples/comm_mode_features_example_argparser.py runs parsing code at top-level (assuming it will be '__main__') - so when the test imports that file it fails.

After this change it succeeds.

Also added `__all__` to a few files so it would succeed cleanly.






[ghstack-poisoned]

aorenste commented:
Superseded by #130497
